extensions/wikia/Listusers/js/table.js

```diff
@@ -0,0 +1,150 @@
+require(['wikia.log', 'wikia.window', 'jquery'], function(log, window, $) {
+    function __makeParamValue() {
+        var f = document.getElementById('lu-form'),
+            target = "",
+            i;
+
+        if (f.lu_target && f.lu_target.length > 0) {
+            for (i = 0; i < f.lu_target.length; i++) {
+                if (f.lu_target[i].checked) {
+                    target += f.lu_target[i].value + ",";
+                }
+            }
+        }
+        return target;
+    }
+
+    // set up data table
+    // @see https://datatables.net/reference/option/ (jquery.dataTables.min.js v1.8.2)
+    var oTable = $('#lu-table').dataTable({
+        "oLanguage": listUsersLanguage,
+        "sCookiePrefix": "Listusers-wikia",
+        "aLengthMenu": [[10, 25, 50], [10, 25, 50]],
+        "sDom": '<"dttoolbar"><"top"flip>rt<"bottom"p><"clear">',
+        "aoColumns": [
+            {"sName": "username"},
+            {"sName": "groups"},
+            {"sName": "revcnt"},
+            {"sName": "dtedit"}
+        ],
+        "bProcessing": true,
+        "bServerSide": true,
+        "bFilter": false,
+        "sPaginationType": "full_numbers",
+        "sAjaxSource": window.wgScript + "?action=ajax&rs=ListusersAjax::axShowUsers",
+        /*"fnInitComplete" : function ( oInstance, oSettings, json ) {
+            // make CSS buttons
+        },*/
+        "fnServerData": function (sSource, aoData, fnCallback) {
+            var limit = 30,
+                offset = 0,
+                groups = __makeParamValue(),
+                loop = 1,
+                order = '',
+                sortingCols = 0,
+                sortColumns = [],
+                sortOrder = [],
+                _tmp = [],
+                columns = [],
+                i;
+
+            for (i in aoData) {
+                log(aoData[i], 'listusers');
+
+                switch (aoData[i].name) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add a "default" clause to this "switch" statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3ASwitchWithoutDefault)

[project-id]: wikia:app